src/azul/__init__.py

```diff
@@ -526,21 +527,24 @@ def __attrs_post_init__(self):
 
         @cached_property
         def is_integration_test_catalog(self) -> bool:
-            return self._it_catalog_re.match(self.name) is not None
-
-        @property
-        def is_internal(self):
-            # FIXME: Remove the second term
-            #        https://github.com/DataBiosphere/azul/issues/2865
-            return self.is_integration_test_catalog or self.name == 'dcp3'
+            if self._it_catalog_re.match(self.name) is not None:
+                require(self.internal,
+                        'IT catalogs must have their `internal` property set to True',
+                        attr.asdict(self))
```

Comment:
Why `asdict`?

Reply:
Provides more context if/when this fires off.

ex. w/ self.internal in the message
```
from azul import config
config.integration_test_catalogs
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/amar/dev/hca/azul/src/azul/caching.py", line 165, in __get__
    value = obj.__dict__[self.fget.__name__] = self.fget(obj)
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 584, in integration_test_catalogs
    return {
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 587, in <dictcomp>
    if catalog.is_integration_test_catalog
  File "/Users/amar/dev/hca/azul/src/azul/caching.py", line 165, in __get__
    value = obj.__dict__[self.fget.__name__] = self.fget(obj)
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 531, in is_integration_test_catalog
    require(self.internal, 'IT catalogs must be internal', self.internal)
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 1148, in require
    reject(not condition, *args, exception=exception)
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 1163, in reject
    raise exception(*args)
azul.RequirementError: ('IT catalogs must be internal', False)
```

vs .asdict
```
from azul import config
config.integration_test_catalogs
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/amar/dev/hca/azul/src/azul/caching.py", line 165, in __get__
    value = obj.__dict__[self.fget.__name__] = self.fget(obj)
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 584, in integration_test_catalogs
    return {
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 587, in <dictcomp>
    if catalog.is_integration_test_catalog
  File "/Users/amar/dev/hca/azul/src/azul/caching.py", line 165, in __get__
    value = obj.__dict__[self.fget.__name__] = self.fget(obj)
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 531, in is_integration_test_catalog
    require(self.internal, 'IT catalogs must be internal', attr.asdict(self))
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 1148, in require
    reject(not condition, *args, exception=exception)
  File "/Users/amar/dev/hca/azul/src/azul/__init__.py", line 1163, in reject
    raise exception(*args)
azul.RequirementError: ('IT catalogs must be internal', {'name': 'it2ebi', 'atlas': 'hca', 'internal': False, 'plugins': {'metadata': {'name': 'hca'}, 'repository': {'name': 'tdr'}}})
```

I suppose adding `self.name, self.internal`  would also provide context, but why not give an operator all information about the catalog with the potential configuration issue?

Reply:
Not what I meant. Why `asdict(self)` and not not just `self` and rely on `repr`>